addons/account/models/account_move.py

```diff
@@ -1175,9 +1175,13 @@ def remove_move_reconcile(self):
             rec_move_ids += account_move_line.matched_credit_ids
         if self.env.context.get('invoice_id'):
             current_invoice = self.env['account.invoice'].browse(self.env.context['invoice_id'])
-            rec_move_ids = rec_move_ids.filtered(
-                lambda r: (r.debit_move_id + r.credit_move_id) & current_invoice.move_id.line_ids
-            )
+            # We only filter out if there is no exchange move created. Indeed, in this case the
+            # exchange move might be reconciled with the payment (if the exchange rate decreased
+            # between invoicing and payment), so we need the payment lines as well.
+            if not current_invoice.payment_move_line_ids.mapped('full_reconcile_id.exchange_move_id'):
```

Comment:
it would work... the other solution would have been to include the exchange move in your filter

```python 
lines_to_keep = current_invoice.move_id.line_ids 
if  current_invoice.move_id.line_ids.mapped('full_reconcile_id.exchange_move_id):
    lines_to_keep += current_invoice.move_id.line_ids.mapped('full_reconcile_id.exchange_move_id.line_ids')
rec_move_ids = rec_move_ids.filtered(
     lambda r: (r.debit_move_id + r.credit_move_id) & lines_to_keep
)
```
I suppose... So you'll end up by removing only the partial reconciliation of the invoice + the exchange rate entry that was linked to the full reconciliation